File: DSA/SakshamKashyap_590014183/day37/day37_Question_2.py
def evaluate_postfix(expression):
    """
    Evaluates a postfix expression using a stack.
    """
    stack = []
    operators = set(['+', '-', '*', '/'])

    for char in expression:
        if char.isdigit():
            # Push numbers onto the stack
            stack.append(int(char))
        elif char in operators:
            # When an operator is found, pop the top two operands
            operand2 = stack.pop()
            operand1 = stack.pop()
            
            # Perform the operation and push the result back
            if char == '+':
                stack.append(operand1 + operand2)
            elif char == '-':
                stack.append(operand1 - operand2)
            elif char == '*':
                stack.append(operand1 * operand2)
            elif char == '/':
                # Integer division
                stack.append(int(operand1 / operand2))
    
    # The final result is the only item left on the stack
    return stack.pop()

# Test cases
print(evaluate_postfix("231*+9-"))
print(evaluate_postfix("52+"))

File: DSA/DivyanshKumarSingh_590012175/Day_37_Question_2.c
#include <stdio.h>
#include <ctype.h>

#define MAX 100

int stack[MAX], top = -1;

void push(int x) { stack[++top] = x; }
int pop() { return stack[top--]; }

int main()
{
    char exp[MAX];
    printf("Enter postfix expression: ");
    scanf("%s", exp);

    for (int i = 0; exp[i] != '\0'; i++)
    {
        if (isdigit(exp[i]))
            push(exp[i] - '0');
        else
        {
            int b = pop();
            int a = pop();
            switch (exp[i])
            {
            case '+':
                push(a + b);
                break;
            case '-':
                push(a - b);
                break;
            case '*':
                push(a * b);
                break;
            case '/':
                push(a / b);
                break;
            }
        }
    }

    printf("Result: %d\n", pop());
    return 0;
}
